datalake/api_stack.py

```diff
@@ -59,7 +60,7 @@ def __init__(  # pylint: disable=too-many-arguments
             if endpoint == "dataset_versions":
                 state_machine_parameter = aws_ssm.StringParameter.from_string_parameter_attributes(
                     self,
-                    "StepFunctionStateMachineARN",
+                    "step-func-statemachine-arn",
```

Comment:
If this *needs* to be the same as the `aws_ssm.StringParameter(` `id` parameter we should make it a constant. Otherwise we should make them different, to avoid confusion. In any case they should be different from the constant value.